server/apiserver/argoserver.go

```diff
@@ -334,7 +334,10 @@ func (as *argoServer) newHTTPServer(ctx context.Context, port int, artifactServe
 		grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(MaxGRPCMessageSize)),
 	}
 	if as.tlsConfig != nil {
-		dialOpts = append(dialOpts, grpc.WithTransportCredentials(credentials.NewTLS(as.tlsConfig)))
+		tlsConfig := as.tlsConfig
+		tlsConfig.InsecureSkipVerify = true
```

Comment:
should you delete this line?

Reply:
This `InsecureSkipVerify` is for communication within the same instance of argo-server process - between grpc-gateway and grpc-server goroutines (since server is serving on `localhost` and we can't have it in certificate SAN for publically trusted certificate). Setting it as `false` will break the gateway.
I provided more reasoning and alternative implementations at https://github.com/argoproj/argo-workflows/pull/9423#discussion_r970628566

Reply:
I think the correct place to do this is here instead
https://github.com/argoproj/argo-workflows/blob/9a32232a85ce0f24719a6aaa46eef86f86a500b4/util/tls/tls.go#L147-L149

Reply:
`func GetServerTLSConfigFromSecret` returns a **server** TLS config with proper certificates. Here we are patching **client** TLS config (dialOpts) for aforementioned reasons. 

We set `InsecureSkipVerify` on server TLS config as well, but this is unnecessary and can become a security risk in the long run.